linera-core/src/chain_worker/state/mod.rs

```diff
@@ -426,6 +426,44 @@ where
         Ok(actions)
     }
 
+    /// Returns true if there are no more outgoing messages in flight up to the given
+    /// block height.
+    pub async fn all_messages_delivered_up_to(
+        &mut self,
+        height: BlockHeight,
+    ) -> Result<bool, WorkerError> {
+        tracing::debug!(
+            "Messages left in {:.8}'s outbox: {:?}",
+            self.chain_id(),
+            self.chain.outbox_counters.get()
+        );
+        let Some((key, _)) = self.chain.outbox_counters.get().first_key_value() else {
+            return Ok(true);
+        };
+        if key > &height {
+            return Ok(true);
+        }
```

Comment:
nit: Can we call `self.chain.all_messages_delivered_up_to` first, then continue with your new code? Perhaps this function can be called `all_messages_delivered_or_intentionally_dropped_up_to` or something like that.